react/dependency installer/e4e1b2e6-26a1-4b3b-b472-4464814fa69a.json

```diff
@@ -0,0 +1,51 @@
+{
+  "implementations": {
+    "es6-node": {
+      "inputFunctions": {
+        "045a62ad-7340-479a-b544-f10d56f16597": "const { exec } = require( \"child_process\" );\nconst { sync } = data;\nconst { dependencies } = params;\nif (data.start !== false ){\n  const command = \"cd \" + data.projectPath + \" && \" + \"npm install \" + params.dependencies.join(' ');\n  outputs.stdout( command, tag );\n  if (state.running){\n    outputs.bounced(data.sync, tag);\n  } else {\n    state.running = true;\n    exec(command, (error, stdout, stderr) => {\n      if (error) {\n        outputs.error(`${error}`, tag);\n      } else {\n        outputs.stdout(stdout || \"\", tag);\n        outputs.stderr(stderr || \"\", tag);\n        outputs.error(null, tag);\n      }\n      state.running = false;\n    });\n  }\n}"
```

Comment:
The statement `const { sync } = data;` will always set the variable `sync` to undefined since data is supposed to contain "projectPath" and "start" as keys.

Reply:
The statement `outputs.bounced(data.sync, tag);` should be `outputs.bounced(data, tag);` since this is the input function of the port `sync` and it will get its value as `data`

Reply:
The parent node's `start` port is supposed to be a signal to start the operation. These kind of ports usually has `any` type and there's no need to validate their value (like here `if (data.start !== false )` since they are simple triggers, if any signal comes in - with whatever value - we start the operation, if we don't want to start, we should not trigger that port. Sending `false` to a start port is somewhat counterintuitive in data flow context imho.

Reply:
> The statement `outputs.bounced(data.sync, tag);` should be `outputs.bounced(data, tag);` since this is the input function of the port `sync` and it will get its value as `data`

changed as suggested

Reply:
> The statement `const { sync } = data;` will always set the variable `sync` to undefined since data is supposed to contain "projectPath" and "start" as keys.

as per previous comment regarding React Installer line has been removed.

Reply:
> The parent node's `start` port is supposed to be a signal to start the operation. These kind of ports usually has `any` type and there's no need to validate their value (like here `if (data.start !== false )` since they are simple triggers, if any signal comes in - with whatever value - we start the operation, if we don't want to start, we should not trigger that port. Sending `false` to a start port is somewhat counterintuitive in data flow context imho.

I think the name start is probably confusing and creates the stated view so I will change it to `run` as I need to control the behaviour given the outcome is system calls.